### PR TITLE
For #18993 - Nimbus: Allow internal tooling to opt into specific branches of an experiment

### LINKED ISSUE #18993: Nimbus: Allow internal tooling to opt into specific branches of an experiment

### PR DESCRIPTION
**Pull request by gabrielluong**
Fixes #18993 

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/nimbus/NimbusBranchesFragment.kt [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        super.onResume()
+        showToolbar(args.experimentName)
+    }
+
+    private fun loadExperimentBranches() {
+        lifecycleScope.launch(Dispatchers.IO) {
+            try {
+                val experiments = requireContext().components.analytics.experiments
+                val branches = experiments.getExperimentBranches(args.experimentId) ?: emptyList()
+                val selectedBranch = experiments.withExperiment(args.experimentId) ?: ""
```
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/nimbus/NimbusBranchesFragment.kt:81]: @jhugman 👀

File: app/src/main/java/org/mozilla/fenix/nimbus/NimbusExperimentsFragment.kt [hunk 2]
```diff
@@ -60,7 +61,7 @@ class NimbusExperimentsFragment : Fragment(R.layout.mozac_service_nimbus_experim
         lifecycleScope.launch(IO) {
             try {
                 val experiments =
-                    requireContext().components.analytics.experiments.getActiveExperiments()
+                    requireContext().components.analytics.experiments.getAvailableExperiments()
```
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/nimbus/NimbusExperimentsFragment.kt:64]: @jhugman 👀

File: app/src/main/java/org/mozilla/fenix/nimbus/controller/NimbusBranchesController.kt [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+ * @param experimentId The string experiment-id or "slug" for a Nimbus experiment.
+ */
+class NimbusBranchesController(
+    private val nimbusBranchesStore: NimbusBranchesStore,
+    private val experiments: NimbusApi,
+    private val experimentId: String
+) : NimbusDetailsAdapterDelegate {
+
+    override fun onBranchItemClicked(branch: Branch) {
+        experiments.optInWithBranch(experimentId, branch.slug)
```
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/nimbus/controller/NimbusBranchesController.kt:29]: @jhugman 👀

**Review by gabrielluong — COMMENTED** [review]
@jhugman Mostly looking for you to review the Nimbus API usage. That I have commented in the code.

**Comment by gabrielluong** [comment]
Still waiting for the AC bump, probably tomorrow.

**Review by jhugman — APPROVED** [review]
This looks fine from a Nimbus POV.

I bow to other's Fenix knowledge.

File: app/src/main/java/org/mozilla/fenix/nimbus/NimbusBranchesFragment.kt [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+                lifecycleScope.launch(Dispatchers.Main) {
+                    nimbusBranchesStore.dispatch(
+                        NimbusBranchesAction.UpdateBranches(
+                            branches,
+                            selectedBranch
+                        )
+                    )
```
Comment (pocmo) [inline app/src/main/java/org/mozilla/fenix/nimbus/NimbusBranchesFragment.kt:90]: Drive-by: There should be no need to switch branches for dispatching. The store has its own internal thread, that it switches to, anyways and observing with `consumeFrom`, like you here, always happens on the main thread.

File: app/src/test/java/org/mozilla/fenix/nimbus/NimbusBranchesStoreTest.kt [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    private lateinit var nimbusBranchesStore: NimbusBranchesStore
+
+    @Before
+    fun setup() {
+        nimbusBranchesState = NimbusBranchesState(branches = emptyList())
+        nimbusBranchesStore = NimbusBranchesStore(nimbusBranchesState)
+    }
+
+    @Test
+    fun testUpdateBranches() = runBlocking {
```
Comment (Amejia481) [inline app/src/test/java/org/mozilla/fenix/nimbus/NimbusBranchesStoreTest.kt:28]: Nit: Should be we follow the same name convention as other tests using `` [backticks](https://kotlinlang.org/docs/coding-conventions.html#names-for-test-methods)?

**Review by Amejia481 — APPROVED** [review]
